File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.java
class Solution {
   
    public boolean validPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;
        
        while(left < right){
            if(s.charAt(left) != s.charAt(right)){
                return PalinChecker(s, left, right - 1) || PalinChecker(s, left + 1, right) ;
            }
            left++;
            right--;
        }
        return true;
    }
    public boolean PalinChecker(String s, int left, int right){
        while(left < right){
            if(s.charAt(left) != s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}

File: 0680-valid-palindrome-ii/0680-valid-palindrome-ii.cpp
class Solution {
public:
    bool validPalindrome(string &s, int l, int r, int count){
        if(count > 1) return false;
        if(l > r) return true;
        
        if(s[l] == s[r]){
            return validPalindrome(s, l+1, r-1, count);
        }else{
            return validPalindrome(s, l, r-1, count+1) || validPalindrome(s, l+1, r, count+1);
        }
    }
    
    bool validPalindrome(string s) {
        return validPalindrome(s, 0, s.size()-1, 0);
    }
};